Adding IPCC AR6 Chapter 3 Fig. 3.9 - Attribution

LisaBock opened the pull request:
## Description

Adding IPCC AR6 Chapter 3 Fig. 3.9 - ANOMALY OF NERA-SURFACE AIR TEMPERATURE - ATTRIBUTION

Original code and a README is provided in the [ESMValTool-AR6-OriginalCode-FinalFigures](https://github.com/ESMValGroup/ESMValTool-AR6-OriginalCode-FinalFigures/blob/main/IPCC_README_files/README_AR6_WG1_Chap3_Figure3_9_surface_temp_attribution.md) repository.

To run this recipe please upload the shapefile of the [AR6 reference regions](https://github.com/SantanderMetGroup/ATLAS/tree/v1.6/reference-regions) ([IPCC-WGI-reference-regions-v4_shapefile.zip](https://github.com/SantanderMetGroup/ATLAS/blob/v1.6/reference-regions/IPCC-WGI-reference-regions-v4_shapefile.zip)), unzip and store it in `<auxiliary_data_dir>/IPCC-regions/` (the `auxiliary_data_dir` is defined in the `config-user.yml` file). The files are already stored in the general `AUX` path in the ESMValTool directory on DKRZ (/work/bd0854/ESMValTool2/AUX/IPCC-Regions/).

Docs: https://esmvaltool--2986.org.readthedocs.build/en/2986/recipes/recipe_ipccwg1ar6ch3.html

* * *

## Checklist

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-title)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#code-quality)
- [x] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#documentation) is available
- [x] [🛠][1] [Tests](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#tests) run successfully
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-checks) were successful

### [New or updated recipe/diagnostic](https://docs.esmvaltool.org/en/latest/community/diagnostic.html)

- [x] [🧪][2] [Recipe runs successfully](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#testing-recipes)
- [x] [🧪][2] [Recipe is well documented](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recipe-and-diagnostic-documentation)
- [x] [🧪][2] [Figure(s) and data](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#diagnostic-output) look as expected from literature
- [x] [🛠][1] [Provenance information](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance) has been added

axel-lauer commented on doc/sphinx/source/recipes/recipe_ipccwg1ar6ch3.rst:
```diff
@@ -35,13 +37,21 @@ following figures from `Eyring et al. (2021)`_ can currently be reproduced:
 .. _`Eyring et al., 2021`: https://www.ipcc.ch/report/ar6/wg1/chapter/chapter-3/
 .. _`Eyring et al. (2021)`: https://www.ipcc.ch/report/ar6/wg1/chapter/chapter-3/
 
+To reproduce Fig. 3.9 please upload the shapefile of the [AR6 reference regions]
+(https://github.com/SantanderMetGroup/ATLAS/tree/v1.6/reference-regions)
+([IPCC-WGI-reference-regions-v4_shapefile.zip]
+(https://github.com/SantanderMetGroup/ATLAS/blob/v1.6/reference-regions/IPCC-WGI-reference-regions-v4_shapefile.zip))
+and store it in `<auxiliary_data_dir>/IPCC-regions/` (the `auxiliary_data_dir`
+is defined in the `config-user.yml` file).
```
I see both, the text and the linked URL including the parentheses when I preview the .rst file. Maybe double-check if the syntax is correct.

axel-lauer commented on doc/sphinx/source/recipes/recipe_ipccwg1ar6ch3.rst:
```diff
@@ -354,6 +388,21 @@ Example plots
+   and 95th percentiles) and for HadCRUT5 (black). All models have been
+   subsampled using the HadCRUT5 observational data mask. Temperatures
+   anomalies are shown relative to 1950-2010 for Antarctica and relative to
+   1850-1900 for other continents. CMIP6 historical simulations are expand by
```
```suggestion
   1850-1900 for other continents. CMIP6 historical simulations are expanded by
```

axel-lauer commented on doc/sphinx/source/recipes/recipe_ipccwg1ar6ch3.rst:
```diff
@@ -354,6 +388,21 @@ Example plots
+   (green), greenhouse gas only (grey) and aerosol only (blue) simulations
+   (multi-model means shown as thick lines, and shaded ranges between the 5th
+   and 95th percentiles) and for HadCRUT5 (black). All models have been
+   subsampled using the HadCRUT5 observational data mask. Temperatures
```
```suggestion
   subsampled using the HadCRUT5 observational data mask. Temperature
```

axel-lauer commented on doc/sphinx/source/recipes/recipe_ipccwg1ar6ch3.rst:
```diff
@@ -354,6 +388,21 @@ Example plots
+   :align:   center
+
+   Figure 3.9: Global, land and ocean annual mean near-surface air
+   temperatures anomalies in CMIP6 models and observations. Timeseries are
```
```suggestion
   temperature anomalies in CMIP6 models and observations. Timeseries are
```

axel-lauer commented on esmvaltool/diag_scripts/ipcc_ar6/tas_anom_damip.ncl:
```diff
@@ -0,0 +1,587 @@
+    statistics = "mean"
+  end if
+
+  caption = "Time series of the " + statistics + " for near-surface " \
+    + "air temperature."
```
Maybe use `long_name` instead of hard-coding the variable name.

axel-lauer commented on esmvaltool/diag_scripts/ipcc_ar6/tas_anom_damip.ncl:
```diff
@@ -0,0 +1,587 @@
+                 domain, \
+                 "times", \
+                 contrib_authors, \
+                 (/"flato13ipcc", "jones13jgr"/), \
```
Shouldn't the AR6 be cited here?

axel-lauer reviewed (changes requested):
I could successfully create the plots, which look like the ones in AR6. That's great! Here a few more small suggestions:

* Additionally create a legend (in a separate output file). Otherwise it is not clear which line corresponds to which forcing (fig 3.9 in AR6 also has a legend).
* The reference 'Iturbide et al. (2020)' mentioned in the caption of the example figure in the recipe documentation (.rst file) could be added to the list of references.
* It would be helpful to add the download information for the shapefile also to the recipe.

LisaBock commented:
Thanks @axel-lauer for reviewing! There were really helpfull comments. 

What I did:
- corrected all typos
- corrected the reference in the diagnostic to the IPCC AR6 Chapter 3
- added a file with the legend
- added the reference 'Iturbide et al. (2020)'  to the recipe and the documentation
- added a mark about the shapefile in the recipe but refer for detailed information to the documentation

remi-kazeroni reviewed (approved):
Nice work, thanks @LisaBock! I could run the recipe successfully with an up-to-date installation of the Tool. This PR looks good to me from a technical point of view. I just have a couple of very minor comments. Once @axel-lauer has approved, this is good to be merged 👍

remi-kazeroni commented:
Thanks for addressing my comments @LisaBock! All looks good to me 👍

axel-lauer commented:
Thanks @LisaBock for taking into account all of my suggestions. This looks all fine with three tiny things left to do regarding the newly added legend:

* it would be great to add the optional `diag_script_info` attribute `legend` to the documentation `recipe_ipccwg1ar6ch3.rst`
* and to the header of the diagnostic script `tas_anom_damip.ncl`
* and (most importantly) to the recipe `recipe_ipccwg1ar6ch3_fig_3_9.yml` so a legend is actually created when running the recipe

LisaBock commented:
Thanks @axel-lauer ! I totally forgot that... It's done now.

axel-lauer commented on doc/sphinx/source/recipes/recipe_ipccwg1ar6ch3.rst:
```diff
@@ -139,6 +156,26 @@ User settings in recipe
+   * y-max: set max of y-axis
+   * header: title for each panel
+   * title: name of region as part of filename
+   * label: set label as output in an extra file
```
```suggestion
   * legend: set labels for optional output of a legend as an extra file
```

axel-lauer commented on esmvaltool/diag_scripts/ipcc_ar6/tas_anom_damip.ncl:
```diff
@@ -0,0 +1,624 @@
+;    y-max: set max of y-axis
+;    header: title for each panel
+;    title: name of region as part of filename
+;    label: set label as ouptu in an extra file
```
```suggestion
;    legend: set labels for optional output of a legend as an extra file
```

axel-lauer commented on esmvaltool/recipes/ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_fig_3_9.yml:
```diff
@@ -0,0 +1,836 @@
+                  "tas_hist-GHG_land", "tas_hist-aer_land"]]
+        reference: ["tasa_ocean", "tasa_global", "tasa_land"]
+        panel_figure_strings: ["a)", "b)", "c)"]
+        label: ["Anthropogenic + Natural", "Natural",
```
```suggestion
        legend: ["Anthropogenic + Natural", "Natural",
```

axel-lauer commented on esmvaltool/recipes/ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_fig_3_9.yml:
```diff
@@ -0,0 +1,836 @@
+                  "tas_hist-GHG_europe", "tas_hist-aer_europe"]]
+        reference: ["tasa_north_america", "tasa_south_america", "tasa_europe"]
+        panel_figure_strings: ["d)", "e)", "f)"]
+        label: ["Anthropogenic + Natural", "Natural",
```
```suggestion
        legend: ["Anthropogenic + Natural", "Natural",
```

axel-lauer commented on esmvaltool/recipes/ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_fig_3_9.yml:
```diff
@@ -0,0 +1,836 @@
+                  "tas_hist-GHG_australasia", "tas_hist-aer_australasia"]]
+        reference: ["tasa_africa", "tasa_asia", "tasa_australasia"]
+        panel_figure_strings: ["g)", "h)", "i)"]
+        label: ["Anthropogenic + Natural", "Natural",
```
```suggestion
        legend: ["Anthropogenic + Natural", "Natural",
```

axel-lauer commented on esmvaltool/recipes/ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_fig_3_9.yml:
```diff
@@ -0,0 +1,836 @@
+                  "tas_hist-GHG_antarctica", "tas_hist-aer_antarctica"]]
+        reference: ["tasa_antarctica"]
+        panel_figure_strings: ["j)"]
+        label: ["Anthropogenic + Natural", "Natural",
```
```suggestion
        legend: ["Anthropogenic + Natural", "Natural",
```

axel-lauer reviewed (changes requested):
I think the diag_script_info attribute is called "legend" in the NCL code...

LisaBock commented:
> I think the diag_script_info attribute is called "legend" in the NCL code...

Yes, of course, it is... Sorry @axel-lauer!
